httpclient5/src/main/java/org/apache/hc/client5/http/protocol/NextNonceInterceptor.java

```diff
@@ -0,0 +1,146 @@
+public class NextNonceInterceptor implements HttpResponseInterceptor {
+
+    public static final HttpResponseInterceptor INSTANCE = new NextNonceInterceptor();
+
+    private final Tokenizer tokenParser = Tokenizer.INSTANCE;
+
+
+    private static final String AUTHENTICATION_INFO_HEADER = "Authentication-Info";
+
+    private static final Tokenizer.Delimiter TOKEN_DELIMS = Tokenizer.delimiters('=', ',');
+    private static final Tokenizer.Delimiter VALUE_DELIMS = Tokenizer.delimiters(',');
+
+    /**
+     * Processes the HTTP response and extracts the {@code nextnonce} parameter from the
+     * {@code Authentication-Info} header if available, storing it in the provided {@code context}.
+     *
+     * @param response the HTTP response containing the {@code Authentication-Info} header
+     * @param entity   the response entity, ignored by this interceptor
+     * @param context  the HTTP context in which to store the {@code nextnonce} parameter
+     * @throws NullPointerException if either {@code response} or {@code context} is null
+     */
+    @Override
+    public void process(final HttpResponse response, final EntityDetails entity, final HttpContext context) {
+        Args.notNull(response, "HTTP response");
+        Args.notNull(context, "HTTP context");
+
+        final Header header = response.getFirstHeader(AUTHENTICATION_INFO_HEADER);
+        if (header != null) {
+            final String nextNonce;
+            if (header instanceof FormattedHeader) {
+                final CharArrayBuffer buf = ((FormattedHeader) header).getBuffer();
+                final ParserCursor cursor = new ParserCursor(0, buf.length());
+                cursor.updatePos(((FormattedHeader) header).getValuePos());
+                nextNonce = parseNextNonce(buf, cursor);
+            } else {
+                final String headerValue = header.getValue();
+                final ParserCursor cursor = new ParserCursor(0, headerValue.length());
+                final CharArrayBuffer buf = new CharArrayBuffer(headerValue.length());
+                buf.append(headerValue);
+                nextNonce = parseNextNonce(buf, cursor);
+            }
+            if (!TextUtils.isBlank(nextNonce)) {
+                //context.setAttribute("auth-nextnonce", nextNonce);
+                HttpClientContext.castOrCreate(context).setNextNonce(nextNonce);
+            }
+        }
+    }
+
+    /**
+     * Parses the {@code Authentication-Info} header content represented by a {@link CharArrayBuffer}
+     * to extract the {@code nextnonce} parameter.
+     *
+     * @param buffer the {@link CharArrayBuffer} containing the value of the {@code Authentication-Info} header
+     * @param cursor the {@link ParserCursor} used to navigate through the buffer content
+     * @return the extracted {@code nextnonce} parameter value, or {@code null} if the parameter is not found
+     */
+    private String parseNextNonce(final CharArrayBuffer buffer, final ParserCursor cursor) {
```

Comment:
@arturobernalg Please use `CharSequence` instead of `CharArrayBuffer` here.

Reply:
done